src/librustc/ty/mod.rs

```diff
@@ -2746,12 +2746,24 @@ fn associated_item_def_ids<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     Rc::new(vec)
 }
 
+fn describe_def<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
+                          def_id: DefId)
+                          -> Option<Def> {
+   let id = tcx.hir.as_local_node_id(def_id).unwrap();
+   let item = tcx.hir.expect_item(id);
+   match item.node{
+       hir::ItemTrait(..) => Some(Def::Trait(def_id)),
+       _ => None
```

Comment:
You are right that was only the traits node I was looking at. There has to be a better way to handle this matching than to hand roll all these matchers... I will just have to leave that to someone more familiar than me.